File: macos/Tests/VPSQuotaCoreTests/AppInfoTests.swift
import Foundation
import Testing
@testable import VPSQuotaCore

@Suite("应用元信息")
struct AppInfoTests {

    @Test("默认回退版本号与运行时版本号有效")
    func versionIsValid() {
        #expect(!AppInfo.fallbackVersion.isEmpty)
        #expect(!AppInfo.version.isEmpty)
        let parts = AppInfo.version.split(separator: ".")
        #expect(parts.count >= 2)
    }

    @Test("Releases 与 Project URL 有效")
    func urlsAreValid() {
        #expect(AppInfo.projectURL.scheme == "https")
        #expect(AppInfo.projectURL.host == "github.com")
        #expect(AppInfo.releasesURL.scheme == "https")
        #expect(AppInfo.releasesURL.absoluteString.hasSuffix("/releases"))
    }
}

File: windows/VpsQuota.Tests/AppInfoTests.cs
namespace VpsQuota.Tests;

using System;
using VpsQuota.Core;
using Xunit;

public class AppInfoTests
{
    [Fact(DisplayName = "版本号非空且符合语义化版本格式")]
    public void VersionIsValid()
    {
        Assert.False(string.IsNullOrWhiteSpace(AppInfo.FallbackVersion));
        Assert.False(string.IsNullOrWhiteSpace(AppInfo.Version));

        var parts = AppInfo.Version.Split('.');
        Assert.True(parts.Length >= 2);
    }

    [Fact(DisplayName = "Releases 与 Project URL 有效")]
    public void UrlsAreValid()
    {
        Assert.Equal("https", AppInfo.ProjectUrl.Scheme);
        Assert.Equal("github.com", AppInfo.ProjectUrl.Host);
        Assert.Equal("https", AppInfo.ReleasesUrl.Scheme);
        Assert.EndsWith("/releases", AppInfo.ReleasesUrl.AbsoluteUri);
    }
}
